docs/en/observability/monitor-k8s/monitor-k8s-application-performance.asciidoc

```diff
@@ -2,52 +2,220 @@
 [[monitor-kubernetes-application-performance]]
 == Part 3: Monitor application performance
 
-[Author: TBD]
+Quickly triage and troubleshoot application performance problems with the help of Elastic
+application performance monitoring (APM).
 
-TODO: Describe how to use APM to monitor applications.
+Think of a latency spike -- APM can help you narrow the scope of your investigation to a single service.
+Because you've also ingested and correlated logs and metrics, you can then link the problem to CPU and memory utilization via the error log entries of a particular Kubernetes pod.
 
 [discrete]
-=== Set up APM Server
+=== Step 1: Set up APM Server
 
-TODO: Describe how to set up APM server.
+Application monitoring data is streamed from your applications running in Kubernetes to APM Server,
+where it is validated, processed, and transformed into {es} documents.
 
-[discrete]
-==== Through ECK
+There are many ways to deploy APM Server when working with Kubernetes.
+This guide assumes you're using our hosted {ess} on {ecloud}.
+If you want to manage APM Server yourself, there are a few alternative options:
+
+* {eck-ref}/[Elastic Cloud on Kubernetes (ECK)] -- The Elastic recommended approach for managing
+APM Server deployed with Kubernetes.
+Built on the Kubernetes Operator pattern, ECK extends basic Kubernetes orchestration capabilities
+to support the setup and management of APM Server on Kubernetes.
+
+* Deploy APM Server as a DaemonSet -- Ensure a running instance of APM Server on each node in your cluster,
+like when a single APM Server instance should be reused for any pod that may be allocated to a node.
+
+* Deploy APM Server as a sidecar -- For environments that should not share an APM Server,
```

Comment:
Fair

Reply:
Hmmm...I think so, especially since we provide a link earlier in the document:

This guide describes how to deploy Elastic monitoring agents as DaemonSets using kubectl and the manifest files from the Beats GitHub repository. For other deployment options, see the Kubernetes operator and custom resource definitions from Elastic Cloud on Kubernetes (ECK) or the Helm charts.

Looking at this ^^ though I wonder if we need to modify the statement to encompass APM.

Reply:
I changed my mind. After previewing the built topics, I don't think you should remove the links.